### PR TITLE
HACK: Make the namespace controller multi-cluster-aware

### PR DESCRIPTION
**Pull request by davidfestal**
The namespace controller should be made multi-cluster-aware, exactly like the CRD controllers have been when working on CRD tenancy (cf. commit 84c07cb)

### CODE REVIEW

**Comment by davidfestal** [comment]
This PR is a followup of PR #11, and will be rebased as soon as the latter is merged.

**Comment by davidfestal** [comment]
This PR fixes issue kcp-dev/kcp#174 on the Kubernetes feature branch side.

**Comment by davidfestal** [comment]
PR rebased on `feature-logical-clusters-1.22` branch after PR #11 has been merged.
@smarterclayton it seems it's ready to review.

**Comment by smarterclayton** [comment]
lgtm

File: pkg/registry/rbac/rest/storage_rbac.go [hunk 1]
```diff
@@ -174,7 +175,9 @@ func (p *PolicyData) EnsureRBACPolicy() genericapiserver.PostStartHookFunc {
 		// initializing roles is really important.  On some e2e runs, we've seen cases where etcd is down when the server
 		// starts, the roles don't initialize, and nothing works.
 		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
-			client, err := clientset.NewForConfig(hookContext.LoopbackClientConfig)
+			adminLogicalClusterConfig := clientgorest.CopyConfig(hookContext.LoopbackClientConfig)
+			adminLogicalClusterConfig.Host += "/clusters/admin"
```
Comment (stevekuznetsov) [inline pkg/registry/rbac/rest/storage_rbac.go:179]: TODO: we should not be hard-coding this as the baseline default, there is a const and builders for that

Comment (davidfestal) [reply pkg/registry/rbac/rest/storage_rbac.go:179]: You're right @stevekuznetsov . Just didn't take the time to extract the const from where it was used. Sorry if it brought problems to you :-(

**Pull request merged**